### PR TITLE
Migration to docusaurus

### PR DESCRIPTION
**Pull request by 0xCaso**
TODO:

- [x] Check if the footer is ok
- [x] Edit the front page 
- [x] Add actions / tests for CI (add publish action after deployment)
- Integrate algolia once deployed

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks a lot. I fixed the footer. Feel free to double-check it. Also the privacy doc. Will work on the landing page next

**Comment by Noc2** [comment]
Screenshot 2023-04-24 at 12 56 20

First landing page draft implemented, see above

File: docusaurus.config.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    // Set the production url of your site here
+    url: 'https://your-docusaurus-test-site.com',
+    // Set the /<baseUrl>/ pathname under which your site is served
+    // For GitHub pages deployment, it is often '/<projectName>/'
+    baseUrl: '/',
+
+    // GitHub pages deployment config.
+    // If you aren't using GitHub pages, you don't need these.
+    organizationName: 'Polkadot', // Usually your GitHub org/user name.
+    projectName: 'docusaurus', // Usually your repo name.
```
Comment (semuelle) [inline docusaurus.config.js]:
```suggestion
    url: 'https://w3f.github.io',
    // Set the /<baseUrl>/ pathname under which your site is served
    // For GitHub pages deployment, it is often '/<projectName>/'
    baseUrl: '/polkadot-spec/',

    // GitHub pages deployment config.
    // If you aren't using GitHub pages, you don't need these.
    organizationName: 'w3f', // Usually your GitHub org/user name.
    projectName: 'polkadot-spec', // Usually your repo name.
```

Good luck. :)

**Comment by Noc2** [comment]
@semuelle Thanks!

**Review by Noc2 — APPROVED** [review]
I'm happy to merge it

**Review by Noc2 — CHANGES_REQUESTED** [review]
@0xCaso One more thing: It probably makes sense to replace "build" with "build_no_kaitai" by default. What do you think?

**Review by Noc2 — APPROVED** [review]

**Pull request merged**